dvc/external_repo.py

```diff
@@ -87,6 +86,30 @@ def clean_repos():
         _remove(path)
 
 
+def _get_remote_config(url):
+    try:
+        repo = Repo(url)
+    except NotDvcRepoError:
+        return {}
+
+    try:
+        name = repo.config["core"].get("remote")
+        if not name:
+            # Fill the empty upstream entry with a new remote pointing to the
+            # original repo's cache location.
+            name = "auto-generated-upstream"
+            return {
+                "core": {"remote": name},
+                "remote": {name: {"url": repo.cache.local.cache_dir}},
+            }
+
+        # Use original remote to make sure that we are using correct url,
+        # credential paths, etc if they are relative to the config location.
+        return {"remote": {name: repo.config["remote"][name]}}
```

Comment:
Previously we only fixed the url, but we actually want paths to keys and credentials too (e.g. there might be something set in original .dvc/config.local, that it makes sense to re-use as well)